src/SignalR/clients/csharp/Client/test/UnitTests/HubConnectionTests.ConnectionLifecycle.cs

```diff
@@ -166,20 +154,43 @@ Task<ConnectionContext> ConnectionFactory(TransferFormat format)
             [Fact]
             public async Task StartAsyncWithFailedHandshakeCanBeStopped()
             {
-                var testConnection = new TestConnection(autoHandshake: false);
-                await AsyncUsing(CreateHubConnection(testConnection), async connection =>
+                var handshakeConnectionErrorLogged = false;
+
+                bool ExpectedErrors(WriteContext writeContext)
                 {
-                    testConnection.Transport.Input.Complete();
-                    try
+                    if (writeContext.LoggerName == typeof(HubConnection).FullName)
                     {
-                        await connection.StartAsync();
+                        if (writeContext.EventId.Name == "ErrorReceivingHandshakeResponse")
+                        {
+                            handshakeConnectionErrorLogged = true;
+                            return true;
+                        }
+
+                        return writeContext.EventId.Name == "ErrorStartingConnection";
                     }
-                    catch
-                    { }
 
-                    await connection.StopAsync();
-                    Assert.True(testConnection.Started.IsCompleted);
-                });
+                    return false;
+                }
+
+                using (StartVerifiableLog(ExpectedErrors))
+                {
+                    var testConnection = new TestConnection(autoHandshake: false);
+                    await AsyncUsing(CreateHubConnection(testConnection, loggerFactory: LoggerFactory), async connection =>
+                    {
+                        testConnection.Transport.Input.Complete();
+                        try
+                        {
+                            await connection.StartAsync();
```

Comment:
nit: OrTimeout

Reply:
Good catch. I cleaned up most of HubConnectionTests.Protocol but missed this test despite modifying it.